homeassistant/components/shelly/config_flow.py

```diff
@@ -50,7 +52,12 @@
     mac_address_from_name,
 )
 
-HOST_SCHEMA: Final = vol.Schema({vol.Required(CONF_HOST): str})
+CONFIG_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_HTTP_PORT): vol.Coerce(int),
```

Comment:
We could also hide this behind `self.show_advanced_options` so it would only be seen if the user has advanced enabled

Reply:
I fear that way most users won't find the feature and complain about it (even if we can state it on the documentation).

Reply:
Most users will not change the `port` value. I think that this field should be hidden by default.